### ISSUE TITLE
Visual glitch with semi transparent particles

### ISSUE THREAD

**Issue by pyrollo** [luanti-org/luanti#10383]
##### Minetest version
```
Minetest 5.3.0 (Linux)
Using Irrlicht 1.8.4
```
##### Summary
When particles with alpha channel textures overlaps, transparency is not properly applied.
Sometimes a foreground particles totally ignores background particles and clear background can be seen through transparent parts:
![thebug](https://user-images.githubusercontent.com/13189280/92933178-36a6a380-f446-11ea-82c9-55049bbc6b20.jpg)

##### Steps to reproduce
Spawn particles with semi transparent PNG textures, in random order back or front to each other. Depending on the z-order, this problem can occur or not.

**Comment by raymoo** [comment]
I think I see the issue. I think Irrlicht uses the position set on the scene nodes for depth sorting, but particles in `updateVertices` update their position by offsetting the vertices of their mesh. I think if particles are modified to translate using `ISceneNode::setPosition` instead, it would sort properly even without disabling depth buffer writes.

**Comment by sorcerykid** [comment]
Yes this is definitely a noticeable problem in my nightclub game as I use dense particle effects, particularly to emulate fog. Since the particles are translucent with the alpha channel, it oftentimes looks wonky. Hopefully an easy solution can be found.

**Comment by pyrollo** [comment]
> I think I see the issue. I think Irrlicht uses the position set on the scene nodes for depth sorting, but particles in `updateVertices` update their position by offsetting the vertices of their mesh. I think if particles are modified to translate using `ISceneNode::setPosition` instead, it would sort properly even without disabling depth buffer writes.

MMhh I did this before (another way that you did) and thought it was useless.

https://github.com/minetest/minetest/pull/10384/commits/ccb80156e5a6f45d80ffdf0e3bfdb631fe00dac2#diff-99eaa43a06f0c2759faf4754351d2301R242

**Comment by raymoo** [comment]
That's a bad place to set it but from the results in #10398 that seems like it should have helped.
